### ISSUE TITLE
`implicit_return` doesn't trigger for if/switch expressions

### ISSUE THREAD

**Issue by lukaskollmer** [realm/SwiftLint#6167]
### Bug Description

The `implicit_return` rule does not trigger in cases where an `if` or `switch` is used as the only expression in a function, and each branch consists of only a single `return` statement:

```swift
// This does not trigger a violation:
func test() -> String {
    if .random() {
        return "A"
    } else {
        return "B"
    }
}
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint
```

### Environment

* SwiftLint version: 0.59.1
* Xcode version: Xcode 16.4 (16F6)
* Installation method used: built from source
* Configuration file:
    ```yml
    enabled_rules:
      - implicit_return
    ```
- no nested configurations
